connection.go

```diff
@@ -418,6 +418,14 @@ func buildSnowflakeConn(ctx context.Context, config Config) (*snowflakeConn, err
 		ctx:             ctx,
 		cfg:             &config,
 	}
+	// do not share ptr to Params map!
+	if sc.cfg.Params != nil {
+		copyParams := make(map[string]*string, len(sc.cfg.Params))
+		for k, v := range sc.cfg.Params {
+			copyParams[k] = v
+		}
+		sc.cfg.Params = copyParams
```

Comment:
I believe this will also need to be mimic'ed in [async requests](https://github.com/snowflakedb/gosnowflake/blob/eaa864cad905d5376097c490a03cb125d38d098b/async.go#L87)